### PR TITLE
legal: Link to Zulip's terms and terms for the active realm

### PR DESCRIPTION
**Pull request by chrisbobbe**
And do a few cleanups/improvements along the way, as usual. 🙂

Greg, I'd really appreciate your help resolving a Flow error in the last commit. It would unlock the ability to bold the realm name where it appears inside a string we want translated.

### CODE REVIEW

File: src/events/eventToAction.js [hunk 1]
```diff
@@ -132,9 +132,28 @@ export default (state: PerAccountState, event: $FlowFixMe): EventAction | null =
...
       };
 
+    case EventTypes.realm:
+      return {
+        type: EVENT,
+        event:
+          /* prettier-ignore */
+          event.op === 'update'
+            // Convert to an equivalent update_dict event, so reducers only have
+            // to handle update_dict.
```
Comment (gnprice) [inline src/events/eventToAction.js:142]: There's also that one property for which the `update` form has a hack where it provides a second property at the same time. If we're not handling that now (which we probably aren't), don't need to handle it in this change, but should leave a TODO comment to do so.

File: src/settings/LegalScreen.js [hunk 2]
```diff
@@ -18,19 +18,35 @@ type Props = $ReadOnly<{|
...
+        // the org itself (e.g., orgs on Zulip Cloud), as this text might be
+        // read to imply. But:
+        // - We don't want to complicate the wording. Not everyone knows
+        //   what a server is.
+        // - These terms will often differ from Zulip's own terms (the ones
+        //   at the other link), and we have to include them.
+        // - These terms will apply to all users in the org, in all cases.
+        label={{
+          text: 'Terms for {realmName}',
+          values: { realmName: <RawLabel style={{ fontWeight: 'bold' }} text={realmName} /> },
```
Comment (gnprice) [inline src/settings/LegalScreen.js:46]: Neat!

From that error message, it looks like the relevant bit of the libdef is here:
```js
  declare export interface IntlFormatters<T = React$Node, R = T> {
// …
    formatMessage(
      descriptor: MessageDescriptor,
      // The `+` was added to make the properties covariant rather
      // than invariant, something TypeScript can't do.
      values?: {| +[key: string]: PrimitiveType | FormatXMLElementFn<string, string> |},
    ): string;
    formatMessage(
      descriptor: MessageDescriptor,
      // The `+` was added to make the properties covariant rather
      // than invariant, something TypeScript can't do.
      values?: {| +[key: string]: PrimitiveType | React$Node | FormatXMLElementFn<T, R> |},
    ): React$Node;
```

So there's an overload -- two signatures for the same method `formatMessage`. Up to now we've used the first one, and now we want to use the second one.

I suspect a key piece of why there's an error here is that, via our `GetText` type, we're saying that our `formatMessage` call (in `TranslationProvider.js`) should return string. From examining those overloaded definitions, one can see that that's not what `formatMessage` returns anymore if you pass it a value that's a `React$Node` -- in that case it returns another `React$Node`.

I'll fiddle with it a bit and see what I find.

**Review by gnprice — COMMENTED** [review]
Thanks! And that's very promising for formatting text in the UI.

I'll spend a few minutes seeing what I can do with the types. One small other comment below.

**Comment by gnprice** [comment]
OK, pushed a couple of commits that I think sort out the types. We end up needing a distinction between two flavors of `LocalizableText`: one to use in a React context, and one to pass to `_` to get a plain string out (e.g. for action sheets or toasts.) Please take a look!

Then some of those changes should go in prep commits, and other parts squashed into the other commits in the branch.

**Comment by chrisbobbe** [comment]
Thanks a lot!!

Writing this down at top-level because it looks like GitHub's iOS app doesn't support commenting directly on lines in a diff: I need to fix the selector for getting the realm description so it gets the realm description, not the realm name.

**Comment by chrisbobbe** [comment]
Note that I haven't yet been able to test the live-updating of the realm name and description. I haven't had time to set up a dev server with Docker lately. I tried to test with the description on CZO by finding a typo and trying to fix it, but the server errored; discussion at https://chat.zulip.org/#narrow/stream/3-backend/topic/502.20on.20updating.20realm.20description/near/1297960.

**Comment by chrisbobbe** [comment]
Pushed a new revision with the `realm_name` -> `name` rename in the realm-update events from #5131.

**Comment by gnprice** [comment]
Thanks! Looks good. Merging with one comment tweak:
```diff
--- a/src/settings/LegalScreen.js
+++ b/src/settings/LegalScreen.js
@@ -35,7 +35,7 @@ export default function LegalScreen(props: Props): Node {
         // These are really terms set by the server admin responsible for
         // hosting the org, and that server admin may or may not represent
         // the org itself, as this text might be read to imply. (E.g.,
-        // that's the case for orgs on Zulip Cloud.) But:
+        // on Zulip Cloud they don't.) But:
```

**Comment by chrisbobbe** [comment]
Thanks!

**Pull request merged**